src/components/o-modal/_template.js

```diff
@@ -0,0 +1,23 @@
+import html from 'nanohtml';
+
+export default ({ open }, childrenFragment) => {
+  const closeModal = () => {
+    const modal = document.getElementById('modal-wrapper');
+    modal.style.display = 'none';
+    modal.classList.add('m-modal--hidden');
+  };
+
+  const closeModalByClickingOutside = (event) => {
+    if (event.target.id === 'modal-wrapper') closeModal();
+  };
+
+  return html`
+    <div id="modal-wrapper" class="m-modal ${open ? '' : 'm-modal--hidden'}" onclick="${closeModalByClickingOutside}">
+      <div class="m-modal__content">
+        <span class="m-modal__close-button"><axa-icon icon="cross-gap" onclick="${closeModal}"></axa-icon></span>
```

Comment:
span is used only for text, not for containing other elements

`Contexts in which this element can be used:
Where phrasing content is expected.`
https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-span-element

Reply:
my i suggest usage of `button` tag instead?